homeassistant/components/emby/media_player.py

```diff
@@ -64,81 +64,45 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Emby platform."""
-
-    host = config.get(CONF_HOST)
-    key = config.get(CONF_API_KEY)
-    port = config.get(CONF_PORT)
-    ssl = config[CONF_SSL]
-
-    if port is None:
-        port = DEFAULT_SSL_PORT if ssl else DEFAULT_PORT
-
-    _LOGGER.debug("Setting up Emby server at: %s:%s", host, port)
-
-    emby = EmbyServer(host, key, port, ssl, hass.loop)
-
-    active_emby_devices: dict[str, EmbyDevice] = {}
-    inactive_emby_devices: dict[str, EmbyDevice] = {}
-
-    @callback
-    def device_update_callback(data):
-        """Handle devices which are added to Emby."""
-        new_devices = []
-        active_devices = []
-        for dev_id, dev in emby.devices.items():
-            active_devices.append(dev_id)
-            if (
-                dev_id not in active_emby_devices
-                and dev_id not in inactive_emby_devices
-            ):
-                new = EmbyDevice(emby, dev_id)
-                active_emby_devices[dev_id] = new
-                new_devices.append(new)
-
-            elif dev_id in inactive_emby_devices and dev.state != "Off":
-                add = inactive_emby_devices.pop(dev_id)
-                active_emby_devices[dev_id] = add
-                _LOGGER.debug("Showing %s, item: %s", dev_id, add)
-                add.set_available(True)
-
-        if new_devices:
-            _LOGGER.debug("Adding new devices: %s", new_devices)
-            async_add_entities(new_devices, True)
-
-    @callback
-    def device_removal_callback(data):
-        """Handle the removal of devices from Emby."""
-        if data in active_emby_devices:
-            rem = active_emby_devices.pop(data)
-            inactive_emby_devices[data] = rem
-            _LOGGER.debug("Inactive %s, item: %s", data, rem)
-            rem.set_available(False)
-
-    @callback
-    def start_emby(event):
-        """Start Emby connection."""
-        emby.start()
-
-    async def stop_emby(event):
-        """Stop Emby connection."""
-        await emby.stop()
-
-    emby.add_new_devices_callback(device_update_callback)
-    emby.add_stale_devices_callback(device_removal_callback)
-
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, start_emby)
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_emby)
+    await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+    )
+
+    ir.async_create_issue(
+        hass,
+        HOMEASSISTANT_DOMAIN,
+        "deprecated_yaml",
+        is_fixable=False,
+        issue_domain=DOMAIN,
+        severity=ir.IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": "Emby",
+        },
+    )
```

Comment:
Continuation on previous comment, if we fail we should raise a different issue. Check `nederlandse_spoorwegen`